code/src/App.js

```diff
@@ -1,9 +1,30 @@
 import React from 'react'
+import { Provider } from 'react-redux'
+import { createStore, combineReducers } from '@reduxjs/toolkit'
 
-export const App = () => {
+import { loadState, saveState } from './localStorage'
+import todos from './reducers/todos'
+import Main from 'components/Main'
+
+const reducer = combineReducers({
+  todos: todos.reducer,
+})
+
+const persistedState = loadState()
+const store = createStore(reducer, persistedState)
+
+store.subscribe(() => {
+  saveState({
+    todos: store.getState().todos
+  })
+})
+
+const App = () => {
```

Comment:
It's great that you included local storage, a todo app without the ability to close the window without losing your to-do's seems pointless